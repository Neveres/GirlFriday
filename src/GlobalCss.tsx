import React from 'react'
import { css, Global } from '@emotion/react'

export const GlobalCss = () => (
  <Global
    styles={css`
      * {
        font-family: ${fontFamily.ubuntu};
        font-style: normal;
        color: ${colors.textWhite};
      }

      body {
        margin: 0;
        background-color: ${colors.bgPrimary};
      }

      #root {
        min-height: 100vh;
        position: relative;
      }

      .mobile-page-title {
        font-weight: ${fontWeight.lighter};
        font-size: ${fontSize.xxLarge};
        line-height: 150%;
        margin-bottom: 24px;
      }
    `}
  />
)

export const fontSize = {
  xxxLarge: '48px',
  xxLarge: '24px',
  xLarge: '16px',
  large: '14px',
  medium: '13px',
  small: '12px',
  xSmall: '11px',
}

export const fontWeight = {
  bolder: 700,
  bold: 600,
  lighter: 400,
}

export const fontFamily = {
  ubuntu: 'Ubuntu',
}

export const colors = {
  bgPrimary: '#181818',
  bgSecondary: '#1B1B1B',
  textWhite: '#ffffff',
  textGray: '#b2b2b2',
}
