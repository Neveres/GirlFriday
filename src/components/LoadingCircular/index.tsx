export { default, default as LoadingCircular } from './LoadingCircular'
