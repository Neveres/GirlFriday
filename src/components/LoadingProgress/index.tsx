export { default, default as LoadingProgress } from './LoadingProgress'
