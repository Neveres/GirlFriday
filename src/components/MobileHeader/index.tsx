export { default, default as MobileHeader } from './MobileHeader'
