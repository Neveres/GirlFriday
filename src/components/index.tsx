export { NavBar } from './NavBar'
export { AppBody } from './AppBody'
export { AppContext } from './AppContext'
export { Input } from './Input'
export { Button } from './Button'
export { Slider } from './Slider'
