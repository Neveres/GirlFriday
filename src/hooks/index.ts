export * from './useInnerWidth'
export * from './useTags'
export * from './useUsers'
export * from './useStorage'
export * from './useFriends'
