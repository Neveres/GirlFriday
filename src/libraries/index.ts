export * from './reducer'
export * from './action'
export * from './httpClient'
export * from './storage'
