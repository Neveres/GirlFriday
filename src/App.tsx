import React, { Suspense, useReducer, useMemo } from 'react'
import { useLocation } from 'react-router'
import { NavBar, AppBody, AppContext } from 'src/components'
import { Routes } from 'src/Routes'
import { useDevice } from 'src/hooks'
import { reducer, SET_MOBILE, SET_SEARCH_PARAMETERS } from 'src/libraries'
import { GlobalCss } from './GlobalCss'

const initialState = {
  isMobile: false,
  searchParameters: {
    page: 1,
    pageSize: 10,
    keyword: '',
  } as GirlFriday.SearchParameters,
}

const App = () => {
  const { pathname } = useLocation()

  const [state, dispatch] = useReducer(reducer, initialState)
  const actions = useMemo(
    () => ({
      setSearchParameters(searchParameters: GirlFriday.SearchParameters) {
        dispatch({ type: SET_SEARCH_PARAMETERS, ...state, searchParameters })
      },
      setMobile(isMobile: boolean) {
        dispatch({ type: SET_MOBILE, ...state, isMobile })
      },
    }),
    [state],
  )
  useDevice({ setMobile: actions.setMobile })
  const valueOfAppContext = useMemo(
    () => ({
      state,
      actions,
    }),
    [state, actions],
  )

  return (
    <>
      <AppContext.Provider value={valueOfAppContext}>
        <NavBar pathname={pathname} />

        <AppBody pathname={pathname}>
          <Suspense fallback={<div>Loading...</div>}>
            <Routes />
          </Suspense>
        </AppBody>
      </AppContext.Provider>

      <GlobalCss />
    </>
  )
}

export default App
