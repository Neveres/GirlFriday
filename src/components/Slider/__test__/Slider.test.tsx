import React from 'react'
import { create } from 'react-test-renderer'
import { Slider } from 'src/components'

describe('Slider', () => {
  const props = {
    setValue: jest.fn(),
    value: 3,
    layout: 'desktop',
  }

  test('should render well under desktop layout', () => {
    const testRenderer = create(<Slider {...props} />)
    expect(testRenderer.toJSON()).toMatchSnapshot()
  })

  test.skip('should render well under mobile layout', () => {
    props.layout = 'mobile'
    const testRenderer = create(<Slider {...props} />)
    expect(testRenderer.toJSON()).toMatchSnapshot()
  })
})
