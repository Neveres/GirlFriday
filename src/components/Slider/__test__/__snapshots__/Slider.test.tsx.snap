// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Slider should render well under desktop layout 1`] = `
<div
  className="css-1g4en5n-Slider"
>
  <span
    className="MuiSlider-root MuiSlider-colorPrimary MuiSlider-sizeMedium MuiSlider-marked css-yw0ea-MuiSlider-root"
    onMouseDown={[Function]}
  >
    <span
      className="MuiSlider-rail css-1e42p1d-MuiSlider-rail"
    />
    <span
      className="MuiSlider-track css-ehpgh7-MuiSlider-track"
      style={
        Object {
          "left": "0%",
          "width": "0%",
        }
      }
    />
    <span
      className="MuiSlider-mark MuiSlider-markActive css-uitvzs-MuiSlider-mark"
      data-index={0}
      style={
        Object {
          "left": "0%",
        }
      }
    />
    <span
      aria-hidden={true}
      className="MuiSlider-markLabel MuiSlider-markLabelActive css-1eoe787-MuiSlider-markLabel"
      data-index={0}
      style={
        Object {
          "left": "0%",
        }
      }
    >
      3
    </span>
    <span
      className="MuiSlider-mark css-ovj1yd-MuiSlider-mark"
      data-index={1}
      style={
        Object {
          "left": "19.16243654822335%",
        }
      }
    />
    <span
      aria-hidden={true}
      className="MuiSlider-markLabel css-yafthl-MuiSlider-markLabel"
      data-index={1}
      style={
        Object {
          "left": "19.16243654822335%",
        }
      }
    >
      6
    </span>
    <span
      className="MuiSlider-mark css-ovj1yd-MuiSlider-mark"
      data-index={2}
      style={
        Object {
          "left": "37.56345177664975%",
        }
      }
    />
    <span
      aria-hidden={true}
      className="MuiSlider-markLabel css-yafthl-MuiSlider-markLabel"
      data-index={2}
      style={
        Object {
          "left": "37.56345177664975%",
        }
      }
    >
      9
    </span>
    <span
      className="MuiSlider-mark css-ovj1yd-MuiSlider-mark"
      data-index={3}
      style={
        Object {
          "left": "56.598984771573605%",
        }
      }
    />
    <span
      aria-hidden={true}
      className="MuiSlider-markLabel css-yafthl-MuiSlider-markLabel"
      data-index={3}
      style={
        Object {
          "left": "56.598984771573605%",
        }
      }
    >
      12
    </span>
    <span
      className="MuiSlider-mark css-ovj1yd-MuiSlider-mark"
      data-index={4}
      style={
        Object {
          "left": "75.1269035532995%",
        }
      }
    />
    <span
      aria-hidden={true}
      className="MuiSlider-markLabel css-yafthl-MuiSlider-markLabel"
      data-index={4}
      style={
        Object {
          "left": "75.1269035532995%",
        }
      }
    >
      15
    </span>
    <span
      className="MuiSlider-mark css-ovj1yd-MuiSlider-mark"
      data-index={5}
      style={
        Object {
          "left": "100%",
        }
      }
    />
    <span
      aria-hidden={true}
      className="MuiSlider-markLabel css-yafthl-MuiSlider-markLabel"
      data-index={5}
      style={
        Object {
          "left": "100%",
        }
      }
    >
      50
    </span>
    <span
      className="MuiSlider-thumb MuiSlider-thumbColorPrimary MuiSlider-thumbSizeMedium css-w5p0kn-MuiSlider-thumb"
      data-focusvisible={false}
      data-index={0}
      onMouseLeave={[Function]}
      onMouseOver={[Function]}
      style={
        Object {
          "left": "0%",
          "pointerEvents": undefined,
        }
      }
    >
      <input
        aria-orientation="horizontal"
        aria-valuemax={50}
        aria-valuemin={3}
        aria-valuenow={3}
        data-index={0}
        disabled={false}
        max={99}
        min={0.5}
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        style={
          Object {
            "border": 0,
            "clip": "rect(0 0 0 0)",
            "direction": "ltr",
            "height": "100%",
            "margin": -1,
            "overflow": "hidden",
            "padding": 0,
            "position": "absolute",
            "whiteSpace": "nowrap",
            "width": "100%",
          }
        }
        type="range"
        value={0.5}
      />
      <span
        aria-hidden={true}
        className="MuiSlider-valueLabel css-nnid7-MuiSlider-valueLabel"
      >
        <span
          className="MuiSlider-valueLabelCircle"
        >
          <span
            className="MuiSlider-valueLabelLabel"
          >
            3
          </span>
        </span>
      </span>
    </span>
  </span>
</div>
`;
